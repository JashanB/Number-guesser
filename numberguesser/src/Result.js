import React from "react";

// const Button = (props) => {
//   return (
//     <button onClick={() => props.setCheckAnswer(state => (false))
//     }
//     >Try Again!</button>
//   )
// }

const Button = (props) => {
  return (
    <button onClick={function () {
      props.setCheckAnswer(state => (false));
      props.setInputNumber(state => 0);
      props.restart ? props.setRestart(state => false) : props.setRestart(state => true)
    }
    }
    >Try Again!</button>
  )
};


export default function Result (props) {
  return (
    <div className="result">
      {props.compareAnswer ? <h3>YOU DID IT! Answer was {props.actualAnswer}</h3> : <h3>Nooo wrong this time. Answer was {props.actualAnswer}</h3>}
      <Button restart={props.restart} setCheckAnswer={props.setCheckAnswer} setRestart={props.setRestart} setInputNumber={props.setInputNumber}/>
    </div>
  )
};