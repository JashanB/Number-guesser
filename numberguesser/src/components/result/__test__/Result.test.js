import React from "react";
import ReactDOM from 'react-dom';
import Result from './../Result';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';

afterEach(cleanup);
it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Result></Result>, div)
});

it ("renders conditional response based on true answer", () => {
  const {getByTestId} = render(<Result compareAnswer={true} actualAnswer={5}></Result>);
  expect(getByTestId('result')).toHaveTextContent("YOU DID IT! Answer was 5");
});

it ("renders conditional response based on false answer", () => {
  const {getByTestId} = render(<Result compareAnswer={false} actualAnswer={5}></Result>);
  expect(getByTestId('result')).toHaveTextContent("Nooo wrong this time. Answer was 5");
});


it("matches snapshot", () => {
  const tree = renderer.create(<Result></Result>).toJSON();
  expect(tree).toMatchSnapshot();
});
