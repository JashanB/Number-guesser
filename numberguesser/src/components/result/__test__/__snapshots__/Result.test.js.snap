// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="result"
  data-testid="result"
>
  <h3>
    Nooo wrong this time. Answer was 
    5
  </h3>
  <button
    data-testid="try-again-button"
    onClick={[Function]}
  >
    Try Again!
  </button>
</div>
`;
